core/src/main/java/com/google/zxing/client/result/WifiResultParser.java

```diff
@@ -36,16 +43,21 @@ public WifiParsedResult parse(Result result) {
     if (!rawText.startsWith("WIFI:")) {
       return null;
     }
-    String ssid = matchSinglePrefixedField("S:", rawText, ';', false);
+    rawText = ";" + rawText.substring("WIFI:".length());
```

Comment:
Why the semicolon in this and later lines?

Reply:
Otherwise there it will also match if one field name is a suffix of another, e.g. if you have

    WIFI:AI:anonidentity;I:identity;...;;

it will set both identity and anonymousIdentity to "anonidentity" (because it only searches for "I:"). I think the best way to avoid that is to require the field name to be directly after a ";" (or to be the first character after the "WIFI:".

Reply:
OK, how about choosing different one-letter prefixes then, like A and H instead of AI and PH2? I know it's kind of cryptic but this is a very short simple format.

Reply:
I don't have any strong opinion on the name of the prefixes, so I'll change it to 'A' and 'H'.

Should the code recognize the prefixes only after a ';' (like it is currently in my pull request) or anywhere (like it is currently in zxing)?